### DISCUSSION TITLE
Blackmagic Ultrastudio Recorder as input device

### DISCUSSION THREAD

**Discussion by somerockenguy** [MindStudioOfficial/livescopestv#2] [Ideas]
Not sure if it is out of the scope of this project but adding support for Blackmagic devices as input devices would be helpful. I find my work split between NDI robo cams and production cameras. Currently I am using NewTek Connect to create a NDI stream from my Blackmagic Ultrastudio Recorder 3G which receives an AUX from our switcher. This works fine but it would be cool to have hardware devices natively supported.

**Comment by MindStudioOfficial** [comment]
Hey, I am currently working on a PRO version of the software that already supports webcams additionally to NDI.

My Implementation uses the more recent Windows Media Foundation (instead of Direct Show). This supports almost all webcam types but some virtual webcam drivers like OBS Virtual Cam or the Blackmagic WDM Capture dont't seem to work properly with this API.

Thats why I chose to additionally implement the Blackmagic DeckLink SDK to support DeckLink devices like the DeckLink PCIe cards or the Thunderbolt Ultrastudio devices.
Implementation is almost done but the release of the PRO version will take some time.
![grafik](https://user-images.githubusercontent.com/62838401/221778553-d62da7ca-7d31-41d7-9b8e-97a253ee0467.png)

**Comment by MindStudioOfficial** [comment]
@somerockenguy I am quite happy to announce that I completed the first milestone of implementing the BlackmagicDesign Decklink SDK in the upcoming PRO version of the software. I was able to use my UltraStudio 3G Recorder as an input device of the software and receive 10BitYUV 4:2:2 video.

![grafik](https://user-images.githubusercontent.com/62838401/224349717-f2efa9ad-274d-48a5-96b7-4a5280d431ee.png)

If you are interested in beta-testing the PRO version let me know :D

**Reply by somerockenguy** [reply]
@MindStudioOfficial I would definitely be interested in checking it on. We have a good amount of BMD gear back at the shop I can test with and report back next week which devices work

**Reply by MindStudioOfficial** [reply]
@somerockenguy That would be awesome. I would do this via E-Mail since I need some info from you and send you the software and license. This way we can stay in close contact and you can report back everything you like.

Hit me up at contact@mindstudio.tv
